File: be/src/main/java/sg/flow/models/user/UserProfile.java
package sg.flow.models.user;

import java.time.LocalDate;

import lombok.Data;

@Data
public class UserProfile {
    
    private String name;
    private String email;
    private String phoneNumber;
    private LocalDate dateOfBirth;
    private String identificationNumber;
    private String address;

    public UserProfile(String name, String email, String phoneNumber, LocalDate dateOfBirth,
            String identificationNumber, String address) {
        this.name = name;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.dateOfBirth = dateOfBirth;
        this.identificationNumber = identificationNumber;
        this.address = address;
    }
}

File: be/src/main/kotlin/sg/flow/models/user/UserProfile.kt
package sg.flow.models.user

import java.time.LocalDate

data class UserProfile(
        val id: Int,
        val name: String,
        val email: String,
        val identificationNumber: String,
        val phoneNumber: String,
        val dateOfBirth: LocalDate,
        val settingJson: String?
)
